public/editor/scripts/project-list.js

```diff
@@ -30,16 +42,54 @@ require(["jquery", "constants", "analytics", "moment"], function($, Constants, a
     return "{{ momentJSLastEdited | safe }}".replace("<% timeElapsed %>", timeElapsed);
   }
 
+  function updateFavorite(projectId, projectSelector, project){
+    var indexOfProjectInFavorites = favorites.indexOf(projectId);
+    var projectFavoriteButton = projectSelector + " .project-favorite-button";
+
+    if(indexOfProjectInFavorites !== -1){
+      favoriteProjectsElementList.push(project);
+      $(projectFavoriteButton).toggleClass("project-favorite-selected");
+      $(projectFavoriteButton).text("Unfavorite");
+    }
+
+    $(projectSelector + " .project-favorite").on("click", function() {
+      indexOfProjectInFavorites = favorites.indexOf(projectId);
```

Comment:
I would probably use a new variable here (simply add a `var` before the variable)